addons/web/controllers/main.py

```diff
@@ -1134,6 +1134,12 @@ def drop(self, master_pwd, name):
         try:
             dispatch_rpc('db','drop', [master_pwd, name])
             request._cr = None  # dropping a database leads to an unusable cursor
+            # if the user is connected to the dropped database, redirect will raise an operational error
+            # trying to access the registry of this database.
+            # Removing db from this request will prevent an invalid error message. (logout looks like a good idea in this case)
+            # https://github.com/odoo/odoo/pull/54102 is proposing a complete fix for this issue.
+            if request.session.db == name:
+                request.session.logout()
```

Comment:
Once #78857 is merged, we'll have to replace `request.session.logout()` by `request._set_cookie_session_id('', '')` to remove the db from the cookie.

Reply:
> Once #78857 is merged, we'll have to replace `request.session.logout()` by `request._set_cookie_session_id('', '')` to remove the db from the cookie.

Not sure I see how that comes into play here?

The point of `logout()` here is to *immediately* clear the `db` ref from the `request` and `session`, so that nothing tries to reuse them now that they're dead. Particularly *during the remaining part of the response generation* for that very `/web/database/drop` request.

Client-side cookies are irrelevant because they will only be sent in a subsequent request, at which point their session doesn't exist anymore, so a new one will be created.